### PR TITLE
Add some option on the scene sidebar

### PR DESCRIPTION
**Pull request by p4535992**
- Adds a menu to the context dropdown for the Scene Directory menus to shut all doors in the selected scene to prepare it for a fresh visit from characters. The Doors that are currently locked remain locked, and are not closed.
- Adds a menu to the context dropdown for the Scene Directory menus to delete fog in the selected scene to prepare it for a fresh visit from characters. I find it useful after QAing a new map for holes in walls/doors and checking lighting, etc.
- Add on the folder scene context options the two options for hide all the scene on the naviagtion bar or show all.

### CODE REVIEW

**Comment by p4535992** [comment]
hi @kandashi sorry to bother, just a reminder of this PR. Let me know if something is not right and i'll try to fix.

**Comment by kandashi** [comment]
Why so many `forEach` updates? this is how you get race conditions.

File: scripts/sidebarContext.js [hunk 1]
```diff
@@ -165,5 +240,54 @@ async function updateChildren() {
...
-}
\ No newline at end of file
+
+}
+
+async function resetDoors(isCurrentScene, id) {
+  if (isCurrentScene) {
+    await canvas
+      .walls?.doors.filter((item) => item.data.ds == 1)
+      .forEach((item) => item.update({ ds: 0 }, {}));
```
Comment (kandashi) [inline scripts/sidebarContext.js:250]: Syncronous updates => race conditions

File: scripts/sidebarContext.js [hunk 2]
```diff
@@ -165,5 +240,54 @@ async function updateChildren() {
...
+  } else {
+    if (id) {
+      await game
+        .scenes?.get(id)
+        ?.data.walls.filter((item) => item.data.door != 0)
+        .forEach((x) => (x.data.ds = 0));
```
Comment (kandashi) [inline scripts/sidebarContext.js:256]: Not actually updates the values in the db, just a local change from what I can tell.

Comment (kandashi) [reply scripts/sidebarContext.js:256]: Replaced with 
```js
async function resetDoors(isCurrentScene, id) {
  let updates = []
  if (isCurrentScene) {
    canvas
      .walls?.doors.filter((item) => item.data.ds === 1)
      .forEach((item) => updates.push({ _id: item.id, ds: 0 }))
    await canvas.scene.updateEmbeddedDocuments("Wall", updates)
  } else {
    if (id) {
      let scene = game.scenes?.get(id)
      scene
        .data.walls.filter((item) => item.data.ds === 1)
        .forEach((x) => updates.push({ _id: x.id, ds : 0 }))
      await scene.updateEmbeddedDocuments("Wall", updates)
    }
  }
  ui.notifications?.info(`Doors have been shut.`);
}
```

File: scripts/sidebarContext.js [hunk 3]
```diff
@@ -165,5 +240,54 @@ async function updateChildren() {
...
+ * @param {boolean} navOn   whether navigation should be on or off for all scenes in the given folder
+ */
+function setNavigationForAllScenes(folder, navOn) {
+  const folderObject = game.folders.get(folder) || game.folders.getName(folder);
+
+  const updates = game.scenes
+      .filter((scene) => scene.data.folder === folderObject.id)
+      .map((scene) => ({ _id: scene.id, navigation: navOn }));
+
+  return Scene.update(updates);
```
Comment (kandashi) [inline scripts/sidebarContext.js:292]: Requires `Scene.updateDocuments(updates)`

**Pull request merged**